grid/social-media/upload-url/profile.supr

```diff
@@ -0,0 +1,222 @@
+
+"""
+Upload Assets via URL to Social Media sites
+Upload images and videos available via an URL, onto social media such as Instagram, Facebook, or Twitter.
+"""
+
+name = "social-media/upload-url"
+version = "1.0.0"
+
+"""
+Upload an Asset
+Register an upload of an asset to a social media profile, in preparation for publishing later.
+"""
+usecase RegisterUpload unsafe {
+  input {
+    """
+    Profile ID
+    Identifier of a profile to upload to. May be optional with some providers.
+    """
+    profileId string!
+
+    """
+    Caption
+    Caption of the asset
+    """
+    caption string!
+
+    """
+    URL
+    URL of the asset. Must be publically acessible.
+    """
+    url! string!
+
+    """
+    Type
+    Media type of the asset
+    """
+    type enum {
+      image,
+      video,
+      reel
+    }
+  }
+
+  result {
+    """
+    Asset ID
+    Identifier of the asset.
+    """
+    assetId! string!
+  }
+
+  error {
+    """
+    Title
+    A short, human-readable summary of the problem type.
+    """
+    title! string
+
+    """
+    Detail
+    A human-readable explanation specific to this occurrence of the problem.
+    """
+    detail string
+
+    """
+    Code
+    Normalized error code usable across providers
+    """
+    code ErrorCode
+
+    """
+    Original Error
+    Original response from the provider
+    """
+    original
+  }
+
+  example Successful {
+    input {
+      profileId = '110658944785122',
+      caption = 'A video of a kitten falling over.',
+      url = 'https://example.com/kitten.mp4',
+    }
+
+    result {
+      assetId = '125682506616099',
+    }
+  }
+
+  example Failed {
+    input {
+      profileId = 'invalid profile ID',
+      caption = 'Caption of the thing.',
+      url = 'https://example.com/kitten.mp4',
+    }
+
+    error {
+      title = 'Bad request'
+    }
+  }
+}
+
+"""
+Get Asset State
```

Comment:
Beside "media", "uploads", and "attachments", you are adding yet another term "asset" which increases confusion. Consider:

- renaming this to sth like GetUploadStatus
- changing attachments in publish-post to assets (or ecistingAssets) or vice versa

Reply:
true, a residue of about the 3rd renaming done :) will fix